### PR TITLE
Feature/mkldnn static

### PR DESCRIPTION
**Pull request by azai91**
currently MKLDNN is dynamically linked to mxnet. This can cause some issues if another version of mkldnn is present on the host machine and mxnet links to the host version. This change statically builds mkldnn into mxnet to handle version issues on linux/mac but keeps it dynamically linked on windows.

removed test_mkldnn_install.py from test as well since it tests if mkldnn is dynamically linked on linux (which it is no longer is)

## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -132,7 +132,12 @@ ifeq ($(USE_MKLDNN), 1)
 		LDFLAGS += -L$(MKLROOT)/lib
 	endif
 	CFLAGS += -I$(MKLDNNROOT)/include
-	LDFLAGS += -L$(MKLDNNROOT)/lib -lmkldnn -Wl,-rpath,'$${ORIGIN}'
+	# MKLDNN but to needs to be dynamically linked for windows as not all VS compilers support static linking
+	ifneq ($(UNAME_S), Windows)
+		LIB_DEP += $(MKLDNNROOT)/lib/libmkldnn.a
+	else
+		LDFLAGS += -L$(MKLDNNROOT)/lib -lmkldnn -Wl,-rpath,'$${ORIGIN}'
```
Comment (ZhennanQin) [inline Makefile]: VS compiler doesn't know those flags. Do we support VS build with Makefile?

Comment (marcoabreu) [reply Makefile]: Nope, windows uses cmake

Comment (ZhennanQin) [reply Makefile]: @marcoabreu Thanks for explanation. This is as I expected. So we don't need to consider Windows build in Makefile.

Comment (ZhennanQin) [reply Makefile]: Why don't remove `LDFLAGS += -L$(MKLDNNROOT)/lib -lmkldnn -Wl,-rpath,'$${ORIGIN}'`? It doesn't work on Windows.

Comment (azai91) [reply Makefile]: The line was there before. I am not trying to change builds on windows.

Comment (ZhennanQin) [reply Makefile]: This line is used for Linux MKLDNN dynamic build. Then why not remove it?

**Pull request merged**